stl/inc/atomic

```diff
@@ -2371,7 +2371,7 @@ public:
         if constexpr (is_always_lock_free) {
             return true;
         } else {
-            return __std_atomic_has_cmpxchg16b() != 0;
+            return _Is_potentially_lock_free && __std_atomic_has_cmpxchg16b() != 0;

```

Comment:
I think it should be `if constexpr` and `else return false`

Reply:
No, because 16-byte `atomic_ref` is lock-free on non-potato chips. (I discovered this because I have changes for a future PR to remove the support for potato chips.) See the earlier test:

https://github.com/microsoft/STL/blob/e36ee6c2b9bc6f5b1f70776c18cf5d3a93a69798/tests/std/tests/P0019R8_atomic_ref/test.cpp#L427-L448

Debug into this:

```
C:\Temp>type meow.cpp
```
```cpp
#include <atomic>
#include <print>
using namespace std;

struct TwoPointers {
    int* left  = nullptr;
    int* right = nullptr;
};

using AtomicRef = atomic_ref<TwoPointers>;

int main() {
    int x = 17;
    int y = 29;
    alignas(AtomicRef::required_alignment) TwoPointers two_pointers{};
    AtomicRef atom_ref{two_pointers};

    println("AtomicRef::is_always_lock_free: {}", AtomicRef::is_always_lock_free);
    println("       atom_ref.is_lock_free(): {}", atom_ref.is_lock_free());
    println("Debug into atom_ref.store():");

    atom_ref.store(TwoPointers{&x, &y});

    println("{} {}", *two_pointers.left, *two_pointers.right);
}
```
```
C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od /Zi /Fdmeow.pdb meow.cpp
meow.cpp

C:\Temp>meow
AtomicRef::is_always_lock_free: false
       atom_ref.is_lock_free(): true
Debug into atom_ref.store():
17 29

C:\Temp>devenv meow.exe meow.cpp
```

It goes through:

https://github.com/microsoft/STL/blob/e36ee6c2b9bc6f5b1f70776c18cf5d3a93a69798/stl/inc/atomic#L1299-L1300

Reply:
I still don't get how this prevent the following interim solution:
```c++
        if constexpr (is_always_lock_free) {
            return true;
        } else if constexpr (_Is_potentially_lock_free)
            return __std_atomic_has_cmpxchg16b() != 0;
        } else {
            return false;
        }
```

Reply:
Oh, that's strictly superior, thanks!